.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,106 @@
+---
+name: unit-test
+
+on:
+  push:
+    paths:
+      - 'src/**'
+      - 'cmake/**'
+      - 'tests/**'
+      - 'conanfile.py'
+      - 'CMakeLists.txt'
+      - '.github/workflows/unit-test.yml'
+    branches:
+      - main
+      - master
+      - dev
+      - 'CURA-*'
+      - 'release/**'
+      - '!15.*'
+      - '![1-4].[0-9]+'
+    tags:
+      - '![1-4].[0-9]+*'
+      - '!Arachne*'
+      - '!ancient*'
+      - '!jedi*'
+      -
+  pull_request:
+    paths:
+      - 'src/**'
+      - 'cmake/**'
+      - 'tests/**'
+      - 'conanfile.py'
+      - 'CMakeLists.txt'
+      - '.github/workflows/unit-test.yml'
+    branches:
+      - main
+      - master
+      - dev
+      - 'release/**'
+
+env:
+  CONAN_USER: ${{ secrets.CONAN_USER }}
+  CONAN_PASS: ${{ secrets.CONAN_PASS }}
+
+jobs:
+  testing:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout CuraEngine
+        uses: actions/checkout@v3
+
+      - name: Setup Python and pip
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.10.4'
+          architecture: 'x64'
+          cache: 'pip'
+
+      - name: Prepare Conan and sip-build (Bash)
```

Comment:
Is it really not possible to re-use steps here? There is a lot of duplication going on here that makes me very anxious...